Birthday-2024-Project/Scripts/ScreenManagement/ScreenLogic.gd

```diff
@@ -0,0 +1,37 @@
+class_name ScreenLogic
+extends Node
+
+signal ScreenEnter
+signal ScreenExit
+
+@export var clearScreenStack : bool = true
+
+var screenManager : ScreenManager
+var transitionData : Dictionary
+
+func ClosePopup(data : Dictionary):
+	screenManager.CloseTopScreen(data)
+	
+func ClosePopupBySignal():
+	screenManager.CloseTopScreen({})
+
+func GoToScreen(screen : PackedScene, data : Dictionary):
+	screenManager.GoToScreen(screen, data)
+
+func ShowSettings():
+	screenManager.ShowSettings()
+
+func ShowConfirmationPopup(title : String, body : String, confirm : String = "OK", cancel : String = "Cancel"):
```

Comment:
this function is unused unless you will modify ExitApplication()

Reply:
The functions on ScreenLogic are mostly just acting as a relay to the ScreenManager. While this function is only currently utilized by the ExitApplication function, it is available for use by anything that might need a two-button popup